Add configurable submodule directories; Add tester.py

arturum1 opened the pull request:
- Add 'dirs' dictonary in meta_data to allow configuring and overriding directories of submodules instead of having a hard-coded fixed path.
- Update functions to use configurable directories (some of the changes are just reverts of previous commits)
- Merge with IObundle/build-lib
- Add tester.py script
- Update ios.py with new auto-map for interfaces
- Included changes I suggested in my review of https://github.com/IObundle/iob-lib/pull/365

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'fixLib' into build-lib

arturum1 committed:
Merge branch 'fixLib' into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'fixLib' into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:arturum1/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'fixLib' into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:arturum1/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
feat(submodule_utils): handle file not found error

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:arturum1/iob-lib into build-lib

arturum1 committed:
feat(tester): add tester.py script
- The tester.py module contains useful functions for the Tester

arturum1 committed:
feat(ios): Add known_map_interfaces
- Dictionary with mappings for each interface

arturum1 committed:
Merge branch 'build-lib' of github.com:arturum1/iob-lib into build-lib

arturum1 committed:
feat(tester.py): add missing code blocks

arturum1 committed:
feat(ios): add prefix to IOs
- Add option to add prefix to every IO signal based on the name of the
  table that contains them. This allows for having multiple tables with
  signals of the same name.
  Multiple tables with duplicate signals often occur when multiple
  instaces of the same peripheral are present, since each of them have
  the same IOs.

arturum1 committed:
feat(setup): add prefix argument for ios

arturum1 committed:
feat(python): fix bugs

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
feat(setup.py): Update directories; Optional generation of Makefile
- Groups all directory related variables into a single 'dirs'
  dictionary.
- Add argument to select if Makefile should be generated on setup. This
  is useful because the setup function of submodules may be called with
  no need for Makefile generation (including config_build.mk file).

AndreMerendeira committed:
[Delivery] added dtest

jjts committed:
Merge pull request #366 from AndreMerendeira/build-lib

[Delivery] added dtest

arturum1 committed:
Merge branch 'build-lib' of github.com:PedroAntunes178/iob-lib into fix

arturum1 committed:
feat(python): add dirs dictionary
- Add 'dirs' dictonary in meta_data to allow configuring and overriding
  directories of submodules instead of having a hard-coded fixed path.
- Update functions to use configurable directories (some of the changes
  are just reverts of previous commits)

arturum1 committed:
Merge branch 'build-lib' of github.com:IObundle/iob-lib into build-lib

arturum1 committed:
Merge branch 'python-setup' into build-lib

PedroAntunes178 commented on scripts/build_srcs.py:
```diff
@@ -9,13 +9,12 @@
 from submodule_utils import import_setup
 import iob_colors
 
-lib_dir = "./submodules/LIB"
```
Why did you decide to remove lib_dir directory?
My idea to have the lib_dir directory global was to make it possible to edit this variable if we wanted to overwrite the directory.

arturum1 replied:
Re-added global lib_dir in 73999b3

PedroAntunes178 commented on scripts/build_srcs.py:
```diff
@@ -9,13 +9,12 @@
     core_version = core_meta_data['version']
     build_dir = core_meta_data['build_dir']
-    core_dir = core_meta_data['core_dir']
+    core_dir = core_meta_data['setup_dir']
```
Why did you choose the name setup_dir?
The setup_dir I interpret it to be the initial directory where the setup.py is called. However, when adding submodules, those should not pass the setup_dir. Instead, they should pass their directory relative to the setup_dir. Which I called core_dir.
What changes did you do to this mechanism?

PedroAntunes178 replied:
This applies to the changes in https://github.com/PedroAntunes178/iob-soc/pull/3 as well.

arturum1 replied:
The method I followed for naming these variables was: build_dir is the directory used to build the core, therefore setup_dir is the directory used to setup the core. 

Renamed all 'core_dir' variables to 'setup_dir' in https://github.com/PedroAntunes178/iob-lib/pull/1/commits/73999b3e8d06cb97a703ed281246fef277dc0311

arturum1 committed:
feat(ios.py): Add color to error message

arturum1 committed:
feat(python): Add IOb-SoC functions
- Add IOb-SoC related functions in setup.py and iob_soc.py (taken from iob_soc_setup.py)
- Re-add lib_dir as a gobal var of build_srcs.py
- Rename core_dir to setup_dir
- Add default directories for submodules
